src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -188,8 +189,26 @@ object WdlExpression {
             }
           case _ => Failure(new WdlExpressionException("Right-hand side of expression must be identifier"))
         }
+      case a: Ast if a.isArrayOrMapIndex =>
+        val index = evaluate(a.getAttribute("rhs"), lookup, functions, interpolateStrings)
+        val mapOrArray = evaluate(a.getAttribute("lhs"), lookup, functions, interpolateStrings)
+        (mapOrArray, index) match {
+          case (Success(a: WdlArray), Success(i: WdlInteger)) =>
+            Try(a.value(i.value)) match {
+              case s:Success[WdlValue] => s
+              case Failure(ex) => Failure(new WdlExpressionException(s"Failed to find index $index on array:\n\n$mapOrArray\n\n${ex.getMessage}"))
+            }
+          case (Success(m: WdlMap), Success(v: WdlValue)) =>
+            m.value.get(v) match {
+              case Some(value) => Success(value)
+              case _ => Failure(new WdlExpressionException(s"Failed to find a key '$index' on a map:\n\n$mapOrArray"))
+            }
+          case (Failure(ex), _) => Failure(ex)
+          case (_, Failure(ex)) => Failure(ex)
+          case (_, _) => Failure(new WdlExpressionException(s"Can't index $mapOrArray with index $index"))
```

Comment:
Oh... so even a.isArrayOrMapIndex doesn't imply the lhs is either a map or array?

(just out of interest, no change required)